python/common/python/math/__init__.java

```diff
@@ -0,0 +1,840 @@
+        } else {
+            return new org.python.types.Float(result);
+        }
+
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v", "w"}
+    )
+    public static org.python.Object hypot(org.python.Object v, org.python.Object w) {
+        double x = toFloatValue(v);
+        double y = toFloatValue(w);
+        double mag = Math.hypot(x, y);
+        if (Double.isInfinite(mag) && !(Double.isInfinite(x) || Double.isInfinite(y))) {
+            throw new org.python.exceptions.OverflowError("math range error");
+        }
+        return new org.python.types.Float(mag);
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object radians(org.python.Object v) {
+        return new org.python.types.Float(Math.toRadians(toFloatValue(v)));
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"v"}
+    )
+    public static org.python.Object degrees(org.python.Object v) {
+        return new org.python.types.Float(Math.toDegrees(toFloatValue(v)));
+    }
+
+
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"arg"}
+    )
+    public static org.python.Object factorial(org.python.Object arg) {
+        double v = 0.0;
+        if (arg instanceof org.python.types.Float) {
+            v = ((org.python.types.Float) arg).value;
+        } else if (arg instanceof org.python.types.Int) {
+            v = (new org.python.types.Float(((org.python.types.Int) arg).value)).value;
+        } else {
+            throw new org.python.exceptions.TypeError("an integer is required (got type " + arg.typeName() + ")");
+        }
+        if ((v - ((int) v)) != 0.0) {
+            throw new org.python.exceptions.ValueError("factorial() only accepts integral values");
+        } else if (v <= 0.0) {
+            throw new org.python.exceptions.ValueError("factorial() not defined for negative values");
+        } else if (v == 0.0 || v == 1.0) {
+            return new org.python.types.Float(1);
+        } else if (v < 0.0 || isnan(v) || isinf(v)) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        } else if (!isIntegral(v)) {
+            throw new org.python.exceptions.ValueError("math domain error");
+        } else {
+            long value = (long) v;
+            BigInteger bi = new BigInteger(Long.toString(value));
+            for (long l = value - 1; l > 1; l--) {         // this implementation doesn't work for large numbers
+                bi = bi.multiply(new BigInteger(Long.toString(l)));  // this needs work
+            }
+            return new org.python.types.Int(bi.intValue());
+        }
+    }
+
+    /*@org.python.Method(
+            __doc__ = "",
+            args = {"arg"}
+    )
+    public static org.python.Object trunc(org.python.Object arg) {
+        return arg.__trunc__();
+    }*/
```

Comment:
It is supposed to be implemented. Since the `__trunc__` method of Int type hadn't been implemented (I don't know if it's done), hence I commented this out, to be left for further discussion.